eShop/eShop/Persistence/Interfaces/IRepository.cs

```diff
@@ -0,0 +1,14 @@
+﻿using System.Linq.Expressions;
+
+namespace eShop.Persistence.Interfaces
+{
+    public interface IRepository<TEntity>
+        where TEntity : class
+    {
+        IQueryable<TEntity> GetAll();
+        IQueryable<TEntity> GetByCondition(Expression<Func<TEntity, bool>> condition);
```

Comment:
I would do this way:
`IRepository<TEntity, TFilter>` and create one method for reading `IQueryable<TEntity> Get(TFilter filter)`